dqn.hpp

```diff
@@ -40,12 +53,14 @@ using NetSp = boost::shared_ptr<caffe::Net<float>>;
 class DQN {
 public:
   DQN(const std::vector<int>& legal_actions,
-      const caffe::SolverParameter& solver_param,
+      const caffe::SolverParameter& solver_actor_param,
+      const caffe::SolverParameter& solver_critic_param,
       const int replay_memory_capacity,
       const double gamma,
       const int clone_frequency) :
         legal_actions_(legal_actions),
-        solver_param_(solver_param),
+        solver_actor_param_(solver_actor_param),
+        solver_critic_param_(solver_critic_param),
```

Comment:
I slightly prefer critic_solver_param to solver_critic_param. Same for actor_solver_param.